#!/usr/bin/env python
"""Config for the visualization of dynamic layers in the scene graph."""
PACKAGE = "hydra_utils"

import dynamic_reconfigure.parameter_generator_catkin as dr_gen  # NOQA

gen = dr_gen.ParameterGenerator()

gen.add(
    "z_offset_scale",
    dr_gen.double_t,
    0,
    "number of steps of offset to apply",
    0.0,
    -5.0,
    10.0,
)
gen.add("visualize", dr_gen.bool_t, 0, "show layer", False)

# Dynamic Layers

gen.add("num_colors", dr_gen.int_t, 0, "number of unique colors to use", 20, 2, 255)
gen.add("color_offset", dr_gen.int_t, 0, "offset into colors", 20, 2, 255)
gen.add(
    "saturation",
    dr_gen.double_t,
    0,
    "saturation for layer color",
    0.8,
    0.0,
    1.0,
)
gen.add(
    "luminance",
    dr_gen.double_t,
    0,
    "luminance for layer color",
    0.8,
    0.0,
    1.0,
)
gen.add(
    "edge_sl_ratio",
    dr_gen.double_t,
    0,
    "ratio of saturation and luminance to node marker color",
    0.6,
    0.0,
    1.0,
)
gen.add(
    "node_scale",
    dr_gen.double_t,
    0,
    "size of the node markers",
    0.1,
    0.01,
    2.0,
)
gen.add(
    "node_alpha",
    dr_gen.double_t,
    0,
    "alpha of the node markers",
    1.0,
    0.0,
    1.0,
)
gen.add(
    "node_use_sphere",
    dr_gen.bool_t,
    0,
    "use sphere markers (instead of cubes) for node centroids",
    False,
)
gen.add(
    "edge_scale",
    dr_gen.double_t,
    0,
    "layer edge size",
    0.03,
    0.001,
    1.0,
)
gen.add("edge_alpha", dr_gen.double_t, 0, "layer edge alpha", 1.0, 0.0, 1.0)
gen.add(
    "label_height",
    dr_gen.double_t,
    0,
    "height of text label above latest node in layer",
    1.0,
    0.0,
    5.0,
)
gen.add(
    "label_scale",
    dr_gen.double_t,
    0,
    "scale of text label above latest node in layer",
    0.5,
    0.05,
    5.0,
)
gen.add(
    "interlayer_edge_insertion_skip",
    dr_gen.int_t,
    0,
    "Number of edges to skip when drawing interlayer edges",
    0,
    0,
    1000,
)
gen.add("visualize_interlayer_edges", dr_gen.bool_t, 0, "show interlayer edges", False)

exit(gen.generate(PACKAGE, PACKAGE, "DynamicLayerVisualizer"))
